django/db/models/sql/query.py

```diff
@@ -89,7 +89,25 @@ def get_children_from_q(q):
             yield from get_paths_from_expression(child)
 
 
+def check_queryset_does_not_contains_outerref(queryset):
+    """
+    https://code.djangoproject.com/ticket/35135#ticket
+
+    In relation to this ticket, we achnowledged we could support passing QuerySets
+    to the condition of a FilteredRelation only if the QuerySet does not contains
+    an OuterRef expression. Instead of building a query that would be wrong,
+    we intentionally prevent such query to be passed and raise an Error instead.
+    """
+    if any(
+        isinstance(where_child.rhs, ResolvedOuterRef)
+        for where_child in queryset.query.where.children
+    ):
+        raise ValueError("Passing OuterRef within a FilteredRelation is not supported.")
```

Comment:
I'm not sure it's worth going this route. Does it catch nested `OuterRef()` calls?